src/plays/typing-speed-test/components/Main.js

```diff
@@ -0,0 +1,181 @@
+        setTimer(60);
+        setIsResultModalOpen(true);
+      }
+    } else {
+      setUserInput(value);
+    }
+  };
+
+  useEffect(() => {
+    if (timer === 0) {
+      setStatus("finished");
+      setIsResultModalOpen(true);
+      return;
+    }
+
+    // To start countdown
+    let timerTimout = setTimeout(() => {
+      if (status === "started") {
+        setTimer(timer - 1);
+      }
+    }, 1000);
+
+    return () => clearTimeout(timerTimout);
+  }, [timer, status]);
+
+  useEffect(() => {
+    // Setting typing text
+    setWords(generateWords());
+
+    // Setting focus on input
+    userInputRef.current.focus();
+  }, []);
+
+  return (
+    <>
+      <div className=" max-w-3xl text-center my-8 mx-auto flex flex-col justify-center">
+        <h2 className="text-[1.6rem] md:text-4xl  text-violet-600 font-bold">
+          Typing Speed Test ⌨️🚀
+        </h2>
+        <p className="text-base my-2 md:text-lg md:my-3 text-gray-600  ">
+          Test your typing skills
+        </p>
+
+        {/* Statistics & Timer */}
+        <div className="flex flex-col justify-around items-center my-5 sm:flex-row md:my-6 md:mt-8 ">
+          <Timer timer={timer} />
+          <Stats
+            wpm={correctWords}
+            cpm={correctCharLength}
+            accuracy={Math.round(
+              (correctWords / (correctWords + incorrectWords)) * 100
+            )}
+          />
+        </div>
+
+        <div className="max-w-3xl my-6 mx-auto leading-9 text-justify md:leading-10 tracking-wide">
+          {words.length ? (
+            words.map((word, index) => (
+              <Word
+                key={index}
+                text={word}
+                isActive={index === activeWordIndex}
+                correct={typedWordsArray[index]}
+              />
+            ))
+          ) : (
+            <p>Text loading please wait ...</p>
+          )}
+        </div>
+
+        <div className="flex items-center justify-center mt-8 space-x-7">
+          <input
+            type="text"
+            ref={userInputRef}
+            disabled={status === "finished"}
+            className="rounded-md border !border-violet-400 !p-3 w-[300px] outline-1 outline-violet-600"
+            placeholder="Start typing..."
+            value={status !== "finished" ? userInput : "Test Completed"}
+            onChange={(e) => processUserInput(e.target.value)}
+          />
+          <div
+            title="Refresh"
+            className="border bg-violet-600 p-3 rounded-md cursor-pointer"
+            onClick={refreshState}
+          >
+            <VscRefresh fontSize={22} color="white" />
+          </div>
+        </div>
+      </div>
+
+      <ResultModal
+        open={isResultModalOpen}
+        handleModalClose={handleModalClose}
+        wpm={correctWords}
+        cpm={correctCharLength}
+        accuracy={Math.round(
+          (correctWords / (correctWords + incorrectWords)) * 100
+        )}
+      />

```

Comment:
```suggestion
      {isResultModalOpen && <ResultModal
        open={isResultModalOpen}
        handleModalClose={handleModalClose}
        wpm={correctWords}
        cpm={correctCharLength}
        accuracy={Math.round(
          (correctWords / (correctWords + incorrectWords)) * 100
        )}
      />}

```

Reply:
As this is a modal component so we should only render it when the open state is true.